cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -72,6 +72,47 @@ message DeleteSurveyResponse {
     bool success = 1;
 }
 
+/**
+    Source Files
+**/
+
+message RegisterSourceSegyFileRequest {
+    Identifier survey = 1;                              // [required] The survey this source file belongs to
+    string path = 2;                                    // [required] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                    // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    ExternalId external_id = 4;                         // [optional] An external identifier - matches service contract field
+    map<string, string> metadata = 5;                   // [optional] Any custom-defined metadata
+    CRS crs = 6;                                        // [required] Official name of the CRS used. Example: "EPSG:23031"
+    SegyOverrides overrides = 7;                        // [optional] Overrides for the source file
+    repeated TraceHeaderField trace_header_fields = 8;  // [optional] The trace header fields that will be used as keys for indexing
+    Dimensions dimensions = 9;                          // [required] File data dimensionality, either 2D or 3D
+}
+
+message RegisterSourceSegyFileResponse {
+    SourceSegyFile file = 1;
+}
+
+message EditSourceSegyFileRequest {
+    Identifier file = 1;                                // [required] The registered source file to edit
+    string path = 2;                                    // [required] Path including protocol, bucket and directory structure. Example: "gs://cognite-seismic-eu/samples/"
+    string name = 3;                                    // [required] Unique filename including extension. Example: "DN1302M03R16_MERGED_KPSDM_00-32_DEG_T.sgy". The name must be unique across buckets and can be used to identify this file in query requests
+    ExternalId external_id = 4;                         // [optional] An external identifier - matches service contract field
+    map<string, string> metadata = 5;                   // [optional] Any custom-defined metadata
+    CRS crs = 6;                                        // [required] Official name of the CRS used. Example: "EPSG:23031"
+    SegyOverrides overrides = 7;                        // [optional] Overrides for the source file
+    repeated TraceHeaderField trace_header_fields = 8;  // [optional] The trace header fields that will be used as keys for indexing
+}
+
+message EditSourceSegyFileResponse {
+    SourceSegyFile file = 1;
+}
+
+message UnregisterSourceSegyFileRequest {
+    Identifier file = 1;                                // [required] The file to unregister
+}
+
+message UnregisterSourceSegyFileResponse {}
```

Comment:
If you look at v1 `DeleteSeismicResponse` and `DeleteSurveyResponse`, they have a `bool succeeded`, so it might be worth it to include that here too for consistency.

On the other hand, it would always be `true` I think, because the failure case will return an error status and not an `UnregisterSourceSegyFileResponse` at all. So maybe not...

Thoughts?

Reply:
I'd side with having the empty message and the failure case being on the error side of the result, we can have a lot more detail in the error message than a true/false